File: Introduction to Programming Languages/Seminars/9. How not to Write Code. Part 3/Extras/Tasks.cs
using System;

namespace Extras
{
    class Tasks
    {
        public static void Task1()
        {
            Console.Clear();
            Console.Write("Задача 1. Дано предложение. Напишите рекурсивный метод, подсчитывающий количество слов в данном предложении. Словом считается последовательность символов без пробелов.\n" +
                          "Введите предложение: ");
            Console.WriteLine($"Количество слов в предложении: {Helpers.CountWords(Console.ReadLine())}");
            Helpers.Hashes();
        }

        public static void Task2() // Неформатированный вывод, печатается быстро.
        {
            Console.Clear();
            Console.WriteLine("Задача 2. Известно, что пароль длиной 3 символа состоит из латинских букв строчного регистра и цифр от 0 до 9. " +
                              "Напишите рекурсивный метод, который перебирает все комбинации паролей.\n");
            char[] options = new char[36];
            Helpers.FillLettersNumbers(options, 0, (char)48, (char)57);
            Helpers.FillLettersNumbers(options, 10, (char)97, (char)122);
            Helpers.GenerateCombinations(options, new char[3], options.Length - 1, 0, 3);
            Console.WriteLine();
            Helpers.Hashes();
        }

        public static void Task2Alt() // Форматированный вывод, печатается долго.
        {
            Console.Clear();
            Console.WriteLine("Задача 2. Известно, что пароль длиной 3 символа состоит из латинских букв строчного регистра и цифр от 0 до 9. " +
                              "Напишите рекурсивный метод, который перебирает все комбинации паролей.\n");
            char[] options = new char[36];
            Helpers.FillLettersNumbers(options, 0, (char)48, (char)57);
            Helpers.FillLettersNumbers(options, 10, (char)97, (char)122);
            Helpers.GenerateCombinationsAlt(options, new char[3], options.Length - 1, 0, 3);
            Console.WriteLine();
            Helpers.Hashes();
        }

        public static void Task3()
        {
            Console.Clear();
            Console.WriteLine("Задача 3. Даны два числа a, b. Сложите их, используя только операции инкремента и декремента.");
            Random random = new Random();
            int a = random.Next(-9, 10), b = random.Next(-9, 10);
            Console.WriteLine($"Сумма чисел {a} и {b} равна {Helpers.SumNumbers(Math.Min(a, b), Math.Max(a, b))}.");
            Helpers.Hashes();
        }

        public static void Task4()
        {
            Console.Clear();
            Console.WriteLine("Задача 4. Даны два числа a, b. Перемножьте их, используя только операции инкремента и декремента.");
            Random random = new Random();
            int a = random.Next(-9, 10), b = random.Next(-9, 10);
            Console.WriteLine($@"Произведение чисел {a} и {b} равно {(a < 0 && b > 0 || a > 0 && b < 0 ?
                              $"-{Helpers.MultiplyNumbers(Math.Abs(a), Math.Abs(b))}" : a == 0 || b == 0 ? "0" : $"{Helpers.MultiplyNumbers(Math.Abs(a), Math.Abs(b))}")}.");
            Helpers.Hashes();
        }
    }
}

File: Introduction to Python/Seminars/3/Homework/tasks.py
import lib


def task1():
    """
    Напишите программу, которая найдёт произведение пар чисел списка. Парой считаем первый и последний элемент, второй и предпоследний и т.д.
    """
    lst = [2, 3, 4, 5, 6]
    print([i * j for i, j in zip(lst[:len(lst) // 2 + 1], lst[len(lst) - 1: len(lst) // 2 - 1: -1])])


def task2():
    """
    Задайте список из вещественных чисел. Напишите программу, которая найдёт разницу между максимальным и минимальным значением дробной части элементов.
    """
    lst = [1.1, 1.2, 3.1, 5.1, 10.01]
    lst = list(map(lambda i: round(i - int(i), 2), lst))
    print(max(lst) - min(lst))


def task3():
    """
    Напишите программу, которая будет преобразовывать десятичное число в двоичное.
    """
    n = int(input("Введите n: "))
    print(f"Число {n} в двоичном представлении: ", end = "")
    lib.bin(n)
    print()


def task4():
    """
    Задайте число. Составьте список чисел Фибоначчи, в том числе для отрицательных индексов.
    """
    n = int(input("Введите n: "))
    print(f"Последовательность НегаФибоначчи для числа {n}: {[lib.fib(i) for i in range(-n, n + 1, 1)]}")


task1()
if (input("\nНажмите Y для перехода к следующему заданию: ").lower() == "y"):
    task2()
if (input("\nНажмите Y для перехода к следующему заданию: ").lower() == "y"):
    task3()
if (input("\nНажмите Y для перехода к следующему заданию: ").lower() == "y"):
    task4()